addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/DwdUnwetterHandler.java

```diff
@@ -0,0 +1,224 @@
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.dwdunwetter.internal.data.DwdWarningsData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DwdUnwetterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Martin Koehler - Initial contribution
+ */
+public class DwdUnwetterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(DwdUnwetterHandler.class);
+
+    private DwdUnwetterConfiguration config;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private DwdWarningsData data;
+
+    public DwdUnwetterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+        }
+    }
+
+    /**
+     * Refreshes the Warning Data.
+     * <p>
+     * The Switch Channel is switched to ON only after all other Channels are updated.
+     * The Switch Channel is switched to OFF before all other Channels are updated.
+     */
+    private void refresh() {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
+        if (!data.refresh()) {
+            return;
+        }
+        int warningCount = getDwdConfig().getWarningCount();
+
+        updateState(getChannelUuid(CHANNEL_LAST_UPDATED), new DateTimeType());
+
+        for (int i = 0; i < warningCount; i++) {
+            State warning = data.getWarning(i);
+            if (warning == OnOffType.OFF) {
+                updateState(getChannelUuid(CHANNEL_WARNING, i), warning);
+            }
+            updateState(getChannelUuid(CHANNEL_SEVERITY, i), data.getSeverity(i));
+            updateState(getChannelUuid(CHANNEL_DESCRIPTION, i), data.getDescription(i));
+            updateState(getChannelUuid(CHANNEL_EFFECTIVE, i), data.getEffective(i));
+            updateState(getChannelUuid(CHANNEL_EXPIRES, i), data.getExpires(i));
+            updateState(getChannelUuid(CHANNEL_ONSET, i), data.getOnset(i));
+            updateState(getChannelUuid(CHANNEL_EVENT, i), data.getEvent(i));
+            updateState(getChannelUuid(CHANNEL_HEADLINE, i), data.getHeadline(i));
+            updateState(getChannelUuid(CHANNEL_ALTITUDE, i), data.getAltitude(i));
+            updateState(getChannelUuid(CHANNEL_CEILING, i), data.getCeiling(i));
+            if (warning == OnOffType.ON) {
+                updateState(getChannelUuid(DwdUnwetterBindingConstants.CHANNEL_WARNING, i), warning);
+            }
+            if (data.isNew(i)) {
+                triggerChannel(getChannelUuid(CHANNEL_UPDATED, i), "NEW");
+            }
+        }
+        data.updateCache();
+    }
+
+    /**
+     * Returns the configuration.
+     *
+     * @return The configuration
+     */
+    private DwdUnwetterConfiguration getDwdConfig() {
```

Comment:
This would cache the config between different `initialize()` calls.
It would probably make sense to simply assign the config in `initialize()` and use the field directly.

Reply:
Done